server-spi/src/main/java/org/keycloak/utils/StringUtil.java

```diff
@@ -56,4 +56,41 @@ public static String joinValuesWithLogicalCondition(String conditionText, Collec
         return options.toString();
     }
 
+    /**
+     * Utility method that substitutes any isWhitespace char to common space ' ' or character 20.
+     * The idea is removing any weird space character in the string like \t, \n, \r.
+     * If quotes character is passed the quotes char is escaped to mark is not the end
+     * of the value (for example escaped \" if quotes char " is found in the string).
+     *
+     * @param str The string to normalize
+     * @param quotes The quotes to escape (for example " or '). It can be null.
+     * @return The string without weird whitespaces and quotes escaped
+     */
+    public static String sanitizeSpacesAndQuotes(String str, Character quotes) {
```

Comment:
Is there some library providing implementation of similar method? If it is OWASP recommendation there could be something like that.

Reply:
I spent some time yesterday and I didn't find anything, so I think there is nothing commonly used. The most closer method was the [normalizeSpace](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#normalizeSpace-java.lang.String-) from commons lang. But I also decided to escape quotes to also cover that part. So I (more or less) reuse the idea to implement the method.